### ISSUE TITLE
Add LoadOrTryCompute method

### ISSUE THREAD

**Issue by mattjohnsonpint** [puzpuzpuz/xsync#150]
#103 describes how to raise an error if `Compute` fails.  In that example, `return 0, true` ensures that the `0` is not added to the map.

That works fine for `Compute`, but I don't see a similar approach possible for `LoadOrCompute` - since the `valueFn` doesn't have a `delete` return value.

So, when using `LoadOrCompute`, how can I ensure that if an error occurs that I don't add a zero value (or `nil`, etc.) to the map?  I wouldn't want a concurrent thread to retrieve that and treat it as an actual good value.

TLDR: Can you please add a version of `LoadOrCompute` that accepts a `valueFn func() (newValue V, delete bool)`?

Thanks.

**Comment by puzpuzpuz** [comment]
`Compute` is flexible enough to be used in `LoadOrCompute` style, so you could use it instead of `LoadOrCompute`:
```go
	counts := xsync.NewIntegerMapOf[int, int]()
	var err error

	v, ok := counts.Compute(42, func(oldValue int, loaded bool) (newValue int, delete bool) {
		if loaded {
			return oldValue, false // return the existing value
		}
		// the error-aware logic goes here
		return
	})
	fmt.Printf("err: %v\n", err)
```

Does this make sense?

**Comment by puzpuzpuz** [comment]
Closing this one as it's been as is for a while. Feel free to comment if something is unclear.

**Comment by mattjohnsonpint** [comment]
Sorry for the delay.  Yes, I can make `Compute` work, if in the case of an error I return `0, true`.  However, if I'm interpreting the comments correctly, `Compute` is taking a full lock with each call.  So there's a perf implication.  I can reduce this by using `Load` first, but that feels a lot like double-checked locking, which I'd like to avoid.  Isn't that the point of `LoadOrCompute`?

**Comment by puzpuzpuz** [comment]
Ok, I understand the use case now: you want to be able to skip adding a computed value. Sounds like you need a new method like `LoadOrComputeIf` or `LoadOrTryCompute`. I'll add it in one of the upcoming versions.

**Comment by mattjohnsonpint** [comment]
`LoadOrTryCompute` sounds perfect. Thanks!
